Java/core/src/main/java/com/amazon/randomcutforest/sampler/SimpleStreamSampler.java

```diff
@@ -102,121 +114,88 @@ public SimpleStreamSampler(int sampleSize, double lambda, long seed) {
+            acceptPointState = new AcceptPointState(sequenceIndex, weight);
+            return true;
         }
-        return Optional.empty();
+        return false;
     }
 
     /**
      * adds the sample to sampler; if the sampler was full, then the sampler has
      * already evicted a point in determining the weight.
      * 
-     * @param point  to be entered in sampler
-     * @param weight computed by acceptSample
-     * @param seqNum timestamp
+     * @param point to be entered in sampler
      */
 
     @Override
-    public void addSample(P point, float weight, long seqNum) {
-        if (storeSequenceIndices) {
-            weightedSamples.add(new Sequential<>(point, weight, seqNum));
-        } else {
-            weightedSamples.add(new Weighted<>(point, weight));
-        }
-        checkState(weightedSamples.size() <= sampleSize,
-                "The number of points in the sampler is greater than the sample size");
+    public void addPoint(P point) {
+        checkState(acceptPointState != null,
+                "this method should only be called after a successful call to acceptSample(long)");
+        samples.add(new Weighted<>(point, acceptPointState.getWeight(), acceptPointState.getSequenceIndex()));
+        acceptPointState = null;
+        checkState(samples.size() <= sampleSize, "The number of points in the sampler is greater than the sample size");
     }
 
     /**
-     * The basic sampling broken down into a proposal/acceptance and a commit. This
-     * breakdown allows the Tree to be in sync with the sampler when duplicates are
-     * present.
-     *
-     * @param point  entry for sample
-     * @param seqNum sequential stamp
-     * @return true if the point is accepted and added, false otherwise
+     * Add a Weighted value directly to the sample. This method is intended to be
+     * used to restore a sampler to a pre-existing state.
+     * 
+     * @param point A weighted point.
      */
-    public boolean sample(P point, long seqNum) {
-        Optional<Float> preSample = acceptSample(seqNum);
-        if (preSample.isPresent()) {
-            addSample(point, preSample.get(), seqNum);
-            return true;
-        }
-        return false;
+    public void addSample(Weighted<P> point) {
+        samples.add(point);
     }
 
     /**
-     * @return the point evicted by the most recent call to {@link #sample}, or null
+     * @return the point evicted by the most recent call to {@link #update}, or null
      *         if no point was evicted.
      */
     @Override
-    public Optional<Sequential<P>> getEvictedPoint() {
+    public Optional<ISampled<P>> getEvictedPoint() {
         return Optional.ofNullable(evictedPoint);
     }
 
     /**
      * @return the list of weighted points currently in the sample. If there is no
      *         sequential information then a dummy variable is placed.
      */
-    @Override
     public List<Weighted<P>> getWeightedSamples() {
```

Comment:
Change to `getWeightedSample` and update comment to reflect the fact that sequence indexes are always stored in SimpleStreamSampler.